ffi/capi/src/fixed_decimal.rs

```diff
@@ -12,6 +12,14 @@ use writeable::Writeable;
 /// Can be obtained via [`icu4x_fixed_decimal_create()`] and destroyed via [`icu4x_fixed_decimal_destroy()`]
 pub type ICU4XFixedDecimal = FixedDecimal;
 
+#[repr(C)]
+/// C-representation version of [`fixed_decimal::Error`]
+pub enum ICU4XFixedDecimalError {
```

Comment:
suggestion: Given that https://github.com/unicode-org/icu4x/pull/770 exists now, should we undo the addition of this error, going back to the old API (for now, until we have a more holistic error story), but use the RTTI stuff for the other results in that file?